src/researchhub/middleware/ApiTokenSession.py

```diff
@@ -0,0 +1,30 @@
+from django.contrib.auth.models import AnonymousUser
+from rest_framework.authentication import TokenAuthentication
+from rest_framework_api_key.permissions import KeyParser
+
+from researchhub.settings import API_KEY_CUSTOM_HEADER
+
+
+def get_user(request):
+    from user.models import UserApiToken
+
+    key_parser = KeyParser()
+    user = None
+    key = key_parser.get(request)
+    try:
+        token = UserApiToken.objects.get_from_key(key)
+        user = token.user
+    except UserApiToken.DoesNotExist:
+        user = AnonymousUser()
```

Comment:
Do we want to allow anonymous user here?
Does this mean anonymous user will be able to create papers?

Reply:
It is okay to have anonymous users here, as it is a part of the whole Django auth process, so it will eventually get caught down the line - meaning that anonymous users won't be able to upload papers because they aren't authenticated. But it probably is better to raise an exception here instead. Good catch.